some improvements to testing (#23)
mongodb/mongo-hibernate: README.md

Both plugins are enabled on gradle's `compileJava` task.

> ### Testing

Reviewer: We could give the CLI command to run all the tests, somewhere in this section

Author: good idea. I added gradle command usage examples for both tests.

> Per best practice, we maintain separate directories for unit and integration testings:

Reviewer: ```suggestion
This project uses separate directories for unit and integration tests:
```

- [unit test](src/test)
- [integration test](src/integrationTest)

> Integration tests will connect to a MongoDB deployment. You may change the default [MongoDB connection string](https://www.mongodb.com/docs/manual/reference/connection-string/) configured as below at [hibernate.properties](src/integrationTest/resources/hibernate.properties):

Reviewer: ```suggestion
Integration tests require a MongoDB deployment. You may change the default [MongoDB connection string](https://www.mongodb.com/docs/manual/reference/connection-string/), below, in [hibernate.properties](src/integrationTest/resources/hibernate.properties):
```

```properties
jakarta.persistence.jdbc.url=mongodb://localhost/mongo-hibernate-test?directConnection=false
```

### CI/CD

> An internal CI/CD pipeline is based on an open-source project [evergreen](https://github.com/evergreen-ci/evergreen), a distributed continuous integration system from MongoDB. The corresponding evergreen configuration resources reside in a [.evergreen](/.evergreen) directory under the project root folder.

Reviewer: ```suggestion
This project uses [evergreen](https://github.com/evergreen-ci/evergreen), a distributed continuous integration system from MongoDB. The evergreen configuration is in the [.evergreen](/.evergreen) directory.
```

## References

> It would be highly helpful to refer to the following awesome articles or resources when need arises:

Reviewer: ```suggestion
```
I don't think we need to comment on this - if the documents have been included under references in the main readme, they must be helpful and good.

Both plugins are enabled on gradle's `compileJava` task.

### Testing

This project uses separate directories for unit and integration tests:

- [unit test](src/test)
- [integration test](src/integrationTest)

#### Gradle Tasks

##### Unit Test
```console
./gradlew clean test
```

##### Integration Test
```console
./gradlew clean integrationTest
```

Integration tests require a MongoDB deployment. You may change the default [MongoDB connection string](https://www.mongodb.com/docs/manual/reference/connection-string/) as below, in [hibernate.properties](src/integrationTest/resources/hibernate.properties):

```properties

> jakarta.persistence.jdbc.url={your_mongodb_connection_string}

Author: I also changed the above properties using placeholder. Repeating existing property seems not to make sense (which is a moving target).
